### PR TITLE
system tests for zqd archive spaces w/ s3

### LINKED ISSUE #975: zqd s3-backed archive spaces system tests
Use ztest and minio to write some system tests for s3 backed archive spaces.

### PR DESCRIPTION
**Pull request by mattnibs**
Write zqd ztests that combine together minio + zqd + zapi. Test for:
1. Running zapi info on an s3-backed archive space
2. Running zapi get on an s3-backed archive space

Closes #975

### CODE REVIEW

File: tests/suite/zqd/archive-s3/services.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+mkdir -p s3/bucket
+mkdir -p zqdroot
+portdir=$(mktemp -d)
+
+
+minio server --writeportfile="$portdir/minio" --quiet --address "localhost:0" ./s3 &
+miniopid=$!
+awaitfile $portdir/minio
+
+# AWS envariables must be set before zqd starts so zqd has access to them.
```
Comment (alfred-landrum) [inline tests/suite/zqd/archive-s3/services.sh:25]: thanks for this comment!

File: tests/suite/zqd/archive-s3/info.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  zar import -s 20KiB -R ./root -data s3://bucket/zartest babble.tzng
+  echo ===
+  zapi -h $ZQD_HOST new -d ./root -k archivestore testsp
+  echo ===
+  zapi -h $ZQD_HOST info testsp | awk '!/id|pcap_path/{print }'
+
+inputs:
+  - name: babble.tzng
+    source: ../../zdx/babble.tzng
+  - name: services
```
Comment (nwt) [inline tests/suite/zqd/archive-s3/info.yaml:12]:
```suggestion
  - name: services.sh
```

File: tests/suite/zqd/archive-s3/services.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    sleep 1
+  done
+}
+
+mkdir -p s3/bucket
+mkdir -p zqdroot
+portdir=$(mktemp -d)
+
+
+minio server --writeportfile="$portdir/minio" --quiet --address "localhost:0" ./s3 &
```
Comment (nwt) [inline tests/suite/zqd/archive-s3/services.sh:21]:
```suggestion
minio server --writeportfile="$portdir/minio" --quiet --address localhost:0 ./s3 &
```

File: tests/suite/zqd/archive-s3/services.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+miniopid=$!
+awaitfile $portdir/minio
+
+# AWS envariables must be set before zqd starts so zqd has access to them.
+export AWS_REGION=does-not-matter
+export AWS_ACCESS_KEY_ID=minioadmin
+export AWS_SECRET_ACCESS_KEY=minioadmin
+export AWS_S3_ENDPOINT=http://localhost:$(cat $portdir/minio)
+
+zqd listen -l="localhost:0" -portfile="$portdir/zqd" -datadir=./zqdroot &
```
Comment (nwt) [inline tests/suite/zqd/archive-s3/services.sh:31]:
```suggestion
zqd listen -l=localhost:0 -portfile="$portdir/zqd" -datadir=./zqdroot &
```

File: tests/suite/zqd/archive-s3/services.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+zqdpid=$!
+trap "rm -rf $portdir; kill -9 $miniopid ; kill -9 $zqdpid" EXIT
```
Comment (nwt) [inline tests/suite/zqd/archive-s3/services.sh:33]:
```suggestion
trap "rm -rf $portdir; kill -9 $miniopid $zqdpid" EXIT
```

**Pull request merged**